### PR TITLE
[BI-38] Create BaseInput component

### PR DESCRIPTION
**Pull request by LAlex14**
image

am facut componenta de input reutilizabila si validarea cu vee validator, in Styleguide, am pus un exemplu, iar pentru verificarea functionalitatii am setat ca la input event sa se afiseze in consola valoarea introdusa, iar cand valoarea din input e valida, butonul Submit devine activ si afiseaza in console textul "hello"

### CODE REVIEW

File: src/plugins/globalComponents.js [hunk 1]
```diff
@@ -1,10 +1,16 @@
-import Vue from 'vue';
 import 'element-ui/lib/theme-chalk/index.css';
 import { Button, Dropdown, Select } from 'element-ui';
+import BaseInput from "@/components/BaseInput.vue";
```
Comment (PascuMihai) [inline src/plugins/globalComponents.js:3]: please use single quotes

File: src/router/index.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ import VueRouter from 'vue-router'
 
 import Home from '@/views/Home.vue'
 import Styleguide from "@/views/Styleguide";
```
Comment (PascuMihai) [inline src/router/index.js:5]: please use single quotes, i know it's annoying but that is the convention :)

File: src/views/Styleguide.vue [hunk 3]
```diff
@@ -1,13 +1,17 @@
 <template>
-  <div></div>
+  <div>
+    <div>{{ $t("styleguide") }}!</div>
```
Comment (PascuMihai) [inline src/views/Styleguide.vue:3]: also here

File: src/views/inputTest.vue [hunk 4]
```diff
@@ -0,0 +1,64 @@
+<template>
+  <div class="w-1/3 px-10 py-5">
+    <h1 class="text-xl font-bold my-5">BaseInput component (email)</h1>
```
Comment (PascuMihai) [inline src/views/inputTest.vue:3]: missed translation

File: src/views/inputTest.vue [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    <ValidationObserver v-slot="{ valid }">
+      <form @submit.prevent="handleSubmit">
+        <base-input
+          v-model="email"
+          type="email"
+          name="email"
+          rules="email|required"
+          placeholder="you@example.com"
```
Comment (PascuMihai) [inline src/views/inputTest.vue:11]: change with Email

File: src/views/inputTest.vue [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+          @input="clgEmail"
+          >Enter email
```
Comment (PascuMihai) [inline src/views/inputTest.vue:13]: missed translation

File: src/views/inputTest.vue (hunk 5 above)
Comment (PascuMihai) [reply src/views/inputTest.vue:11]: Also translation

File: src/views/inputTest.vue [hunk 7]
```diff
@@ -0,0 +1,64 @@
+<template>
+  <div class="w-1/3 px-10 py-5">
+    <h1 class="text-xl font-bold my-5">BaseInput component (email)</h1>
+    <ValidationObserver v-slot="{ valid }">
+      <form @submit.prevent="handleSubmit">
+        <base-input
+          v-model="email"
+          type="email"
+          name="email"
```
Comment (PascuMihai) [inline src/views/inputTest.vue:9]: add translation, also to the other same fields

File: src/views/inputTest.vue [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+          rules="email|required"
+          placeholder="you@example.com"
+          @input="clgEmail"
+          >Enter email
+        </base-input>
+        <button
+          type="submit"
+          class="
```
Comment (PascuMihai) [inline src/views/inputTest.vue:17]: just format the props on new lines, the css classes can be on the same line until it reaches the edge of the screen

File: src/views/inputTest.vue [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+            font-medium
+            text-white
+            bg-indigo-600
+            hover:bg-indigo-700
+            focus:outline-none
+            focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500
+          "
+          :disabled="!valid"
+        >
+          Submit
```
Comment (PascuMihai) [inline src/views/inputTest.vue:37]: translation

File: src/views/inputTest.vue [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    </ValidationObserver>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "inputTest",
+  data() {
+    return {
+      email: "",
```
Comment (PascuMihai) [inline src/views/inputTest.vue:49]: single quotes

File: src/views/inputTest.vue [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+          Submit
+        </button>
+      </form>
+    </ValidationObserver>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "inputTest",
```
Comment (PascuMihai) [inline src/views/inputTest.vue:46]: This component should be moved tho the styleguide page, there we will have all the base compoents to test them

File: src/views/inputTest.vue [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+  data() {
+    return {
+      email: "",
+    };
+  },
+  methods: {
+    handleSubmit() {
+      console.log("hello");
+    },
+    clgEmail() {
```
Comment (PascuMihai) [inline src/views/inputTest.vue:56]: as a common rule we need to use clear names for variables and methods, so for this one should be called onInput

Comment (PascuMihai) [reply src/views/inputTest.vue:56]: To describe the action of the purpose for that name

File: src/components/BaseInput.vue [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+          :class="
+            errors.length
+              ? 'pr-10 border-red-300 text-red-900 placeholder-red-300 focus:outline-none focus:ring-red-500 focus:border-red-500'
+              : 'shadow-sm focus:ring-indigo-500 focus:border-indigo-500 border-gray-300'
+          "
+          v-bind="$attrs"
+        />
+        <div
+          v-show="errors.length"
+          class="
```
Comment (PascuMihai) [inline src/components/BaseInput.vue:23]: let's formt the classes on line until it reaches the edge of the screen

File: package.json [hunk 14]
```diff
@@ -8,20 +8,30 @@
     "lint": "vue-cli-service lint"
   },
   "dependencies": {
+    "@tailwindcss/forms": "^0.3.3",
+    "@vue-hero-icons/solid": "^1.7.2",
```
Comment (PascuMihai) [inline package.json:12]: We don't need to install a packge just for an icon, let's uninstall this and maybe find an svg or don't put anything at all

File: src/components/BaseInput.vue [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+        >
+          {{ error }}
+        </p>
+      </div>
+    </validation-provider>
+  </div>
+</template>
+
+<script>
+import { ExclamationCircleIcon } from "@vue-hero-icons/solid";
```
Comment (PascuMihai) [inline src/components/BaseInput.vue:53]: Let's remove this

**Review by PascuMihai — APPROVED** [review]
Nice work 👍  approved

**Review by Ceciliaa — APPROVED** [review]

**Pull request merged**